### PR TITLE
Umang

### PR DESCRIPTION
**Pull request by gucci1909**
middleware , redux token in api's and Basic Item Recommendation Algorithm

### CODE REVIEW

File: client/store/destinationStore.js [hunk 1]
```diff
@@ -6,7 +6,11 @@ export const processGeoJSON = createAsyncThunk(
   "destination/processGeoJSON",
   async (data, { rejectWithValue }) => {
     try {
-      const response = await axios.post(`${api}/osm/process/geojson`, data);
+      const token = await AsyncStorage.getItem('userToken');
```
Comment (andrew-bierman) [inline client/store/destinationStore.js:9]: We should just add a middleware to our store like this, would cut down on a top of repeated code. Rough idea:

`const authMiddleware = (storeAPI) => (next) => (action) => {
  if (action.type.includes('pending')) {
    const token = storeAPI.getState().auth.token;

    if (token) {
      // Attach token to headers
      action.meta.arg.headers = { ...action.meta.arg.headers, Authorization: `Bearer ${token}` };
    }
  }

  return next(action);
};
`

**Review by andrew-bierman — CHANGES_REQUESTED** [review]
See comments

**Comment by andrew-bierman** [comment]
I'm getting 'invalid token' error for all API requests. Are you experiencing same thing ?

**Comment by gucci1909** [comment]
> I'm getting 'invalid token' error for all API requests. Are you experiencing same thing ?

yes @andrew-bierman same

**Comment by andrew-bierman** [comment]
These should be split into different pull requests

**Pull request closed without merging**